src/reader/ArticleReader.js

```diff
@@ -152,16 +161,25 @@ export default function ArticleReader({ api, teacherArticleID }) {
 
     window.addEventListener("focus", handleFocus);
     window.addEventListener("blur", handleBlur);
-    window.addEventListener("scroll", handleScroll);
+    // https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#usecapture
+    window.addEventListener("scroll", handleScroll, true);
+    window.addEventListener("beforeunload", componentWillUnmount);
   }
 
   function componentWillUnmount() {
     uploadActivity();
+    /*
```

Comment:
can we safely delete this?

Reply:
Delete what exactly? I decided to not comment out this yet to avoid too much logging to the database while testing. 

@mircealungu There is one thing I noticed. Depending on resolution, the viewport can be much longer for some users than others. We track the TOP scroll position, and I am wondering if we should also track how long the viewPort is so we can estimate the section they saw. What do you think?